engine/src/main/java/com/redhat/cloud/notifications/routers/DailyDigestResource.java

```diff
@@ -0,0 +1,74 @@
+package com.redhat.cloud.notifications.routers;
+
+import com.redhat.cloud.notifications.events.KafkaMessageWithIdBuilder;
+import com.redhat.cloud.notifications.models.AggregationCommand;
+import com.redhat.cloud.notifications.models.EmailAggregationKey;
+import com.redhat.cloud.notifications.models.EmailSubscriptionType;
+import com.redhat.cloud.notifications.models.Environment;
+import com.redhat.cloud.notifications.routers.dailydigest.TriggerDailyDigestRequestDto;
+import io.vertx.core.json.Json;
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.Emitter;
+import org.eclipse.microprofile.reactive.messaging.Message;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Response;
+
+import static com.redhat.cloud.notifications.Constants.API_INTERNAL;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.MediaType.TEXT_PLAIN;
+
+@Path(API_INTERNAL + "/daily-digest")
+public class DailyDigestResource {
+
+    public static final String AGGREGATION_OUT_CHANNEL = "toaggregation";
+
+    @Channel(AGGREGATION_OUT_CHANNEL)
+    @Inject
+    Emitter<String> aggregationEmitter;
+
+    @Inject
+    Environment environment;
+
+    /**
+     * Triggers a daily digest by sending a command to the Kafka "aggregation"
+     * queue.
+     * @param triggerDailyDigestRequestDto the settings of the aggregation
+     *                                     command.
+     */
+    @Consumes(APPLICATION_JSON)
+    @Produces(TEXT_PLAIN)
+    @POST
+    public Response triggerDailyDigest(final TriggerDailyDigestRequestDto triggerDailyDigestRequestDto) {
+        if (!this.environment.isEnvironmentStage()) {
+            return Response
+                .status(Response.Status.BAD_REQUEST)
+                .entity("the daily digests can only be triggered in the stage environment")
+                .type(TEXT_PLAIN)
+                .build();
+        }
+
+        final EmailAggregationKey emailAggregationKey = new EmailAggregationKey(
+            triggerDailyDigestRequestDto.getOrgId(),
+            triggerDailyDigestRequestDto.getBundleName(),
+            triggerDailyDigestRequestDto.getApplicationName()
+        );
+
+        final AggregationCommand aggregationCommand = new AggregationCommand(
+            emailAggregationKey,
+            triggerDailyDigestRequestDto.getStart(),
+            triggerDailyDigestRequestDto.getEnd(),
+            EmailSubscriptionType.DAILY
+        );
+
+        final Message<String> message = KafkaMessageWithIdBuilder.build(Json.encode(aggregationCommand));
```

Comment:
I would use an `ObjectMapper` here to serialize the command instead of `Json.encode` to make sure this class behaves consistently with the `aggregator` module.

Reply:
`KafkaMessageWithIdBuilder` isn't really needed as we don't consume the ID on the other side of the Kafka topic, but it doesn't hurt so it can stay.

Reply:
All right, applied the suggestion!